### ISSUE TITLE
Warnings on unused fields not triggered when class is abstract

### ISSUE THREAD

**Issue by TobiasBreuer** [dotnet/roslyn#13652]
**Version Used**: VS 2015 Update 3

**Steps to Reproduce**:

When having an `abstract` class containing `private` fields that are never assigned to or never read, the native compiler (VS 2012) used to issue related warnings CS0414 / CS0649. The roslyn compiler in VS 2015 however does only trigger these warnings when the class is not being marked as `abstract`.

For details please see my related [StackOverflow question](http://stackoverflow.com/questions/39324517/visual-studio-2015-warnings-not-triggered).

For a better overview, I add the minimal example on how to reproduce this issue here as well:

```csharp
    // When the class is being marked as abstract, VS-2015 will not issue any warning.
    // When the class is not abstract the warnings are issued.
    public abstract class Demonstrator
    {
        // This should trigger warning CS0169 Warning Level 2
        private int _UnusedField;

        // This should trigger warning CS0414 Warning Level 4
        private int _AssignedButNotReadField;

        public Demonstrator()
        {
            _AssignedButNotReadField = 1;
        }
    }
```

The compiler should still trigger those warnings even though the class is abstract, as, due to the fact that those fields are private, their usage is still restricted to that single class.
